src/actinia_core/core/common/process_chain.py

```diff
@@ -438,38 +506,12 @@ def _create_module_process(self, module_descr):
 
         stdin_func = None
         if "stdin" in module_descr:
-            if "::" not in module_descr["stdin"]:
-                raise AsyncProcessError(
-                    "The stdin option in id %s misses the ::" % str(id))
-
-            object_id, method = module_descr["stdin"].split("::")
-
-            if "stdout" == method is True:
-                stdin_func = self.process_dict[object_id].stdout
-
-            elif "stderr" == method is True:
-                stdin_func = self.process_dict[object_id].stderr
-            else:
-                raise AsyncProcessError(
-                    "The stdout or stderr flag in id %s is missing" % str(id))
+            stdin_func = self._create_stdin_process(module_descr, id)
 
         # Check for stdout parser that can be of type table, list or key/value pairs
         # and store the definition in a list
         if "stdout" in module_descr:
-            if "id" not in module_descr["stdout"]:
-                raise AsyncProcessError(
-                    "Missing unique *id* in stdout parser description of process id %s"
-                    % str(id))
-            if "format" not in module_descr["stdout"]:
-                raise AsyncProcessError(
-                    "Missing *format* in stdout parser description of process id %s"
-                    % str(id))
-            if "delimiter" not in module_descr["stdout"]:
-                raise AsyncProcessError(
-                    "Missing *delimiter* in stdout parser description of process id %s"
-                    % str(id))
-
-            self.output_parser_list.append({id: module_descr["stdout"]})
+            self._check_stdout(module_descr, id)
```

Comment:
Maybe rename to `_check_and_add_stdout` ? Or even `_check_and_get_stdout_command`  to make it similar to the other ones if you think it fits.